File: Data Structures & Algorithms/dynamicArray/submission-1.go
type DynamicArray struct {
    arr []int
    capacity int
    size int
}

func NewDynamicArray(capacity int) *DynamicArray {
    da := DynamicArray{}
    da.arr = make([]int,capacity)
    da.capacity = capacity
    da.size = 0
    return &da
}

func (da *DynamicArray) Get(i int) int {
    return da.arr[i]
}

func (da *DynamicArray) Set(i int, n int) {
    da.arr[i] = n
}

func (da *DynamicArray) Pushback(n int) {
    if da.size == da.capacity {
        da.resize()
    }
    da.arr[da.size] = n
    da.size += 1
}

func (da *DynamicArray) Popback() int {
    da.size -= 1
    return da.arr[da.size]
}

func (da *DynamicArray) resize() {
    da.capacity *= 2
    arr := make([]int,da.capacity)
    for i,n := range da.arr {
        arr[i] = n
    }
    da.arr = arr
}

func (da *DynamicArray) GetSize() int {
    return da.size
}

func (da *DynamicArray) GetCapacity() int {
    return da.capacity
}

File: Data Structures & Algorithms/hashTable/submission-1.py
class Pair:
    def __init__(self, key: int, val: int) -> None:
        self.key = key
        self.val = val

class HashTable:
    def __init__(self, capacity: int):
        self.capacity = capacity
        self.size = 0
        self.map = [None for _ in range(capacity)]
    
    def hash(self, key:int) -> None:
        return key % self.capacity

    def insert(self, key: int, value: int) -> None:
        idx = self.hash(key)
        while True:
            if self.map[idx] == None:
                self.map[idx] = Pair(key,value)
                self.size += 1
                if self.size >= self.capacity//2:
                    self.resize()
                break
            elif self.map[idx].key == key:
                self.map[idx].val = value
                break
            idx = (idx+1)%self.capacity

    def get(self, key: int) -> int:
        idx = self.hash(key)
        while self.map[idx] != None:
            if self.map[idx].key == key:
                return self.map[idx].val
            idx = (idx+1)%self.capacity
        return -1

    def remove(self, key: int) -> bool:
        idx = self.hash(key)
        while self.map[idx] != None:
            if self.map[idx].key == key:
                self.map[idx] = None
                self.size -= 1
                return True
            idx = (idx+1)%self.capacity
        return False

    def getSize(self) -> int:
        return self.size

    def getCapacity(self) -> int:
        return self.capacity

    def resize(self) -> None:
        self.capacity *= 2
        oldMap = self.map
        self.map = [None for _ in range(self.capacity)]
        self.size = 0
        for pair in oldMap:
            if pair:
                self.insert(pair.key,pair.val)
